dataland-user-service/src/main/kotlin/org/dataland/datalanduserservice/service/PortfolioBulkDataRequestService.kt

```diff
@@ -0,0 +1,135 @@
+package org.dataland.datalanduserservice.service
+
+import org.dataland.datalandcommunitymanager.openApiClient.api.RequestControllerApi
+import org.dataland.datalandcommunitymanager.openApiClient.model.BulkDataRequest
+import org.dataland.datalanduserservice.model.BasePortfolio
+import org.dataland.datalanduserservice.model.EnrichedPortfolio
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+/**
+ * Service to create Bulk Data Requests upon Portfolio or Monitoring changes.
+ */
+@Service
+class PortfolioBulkDataRequestService
+    @Autowired
+    constructor(
+        private val requestControllerApi: RequestControllerApi,
+        private val portfolioEnrichmentService: PortfolioEnrichmentService,
+    ) {
+        /**
+         * Sends Bulk Data Requests with respect to the Monitoring Status and Company Sectors.
+         */
+        fun sendBulkDataRequestIfMonitored(portfolio: BasePortfolio) {
+            if (portfolio.isMonitored) {
+                val enrichedPortfolio = portfolioEnrichmentService.getEnrichedPortfolio(portfolio)
+                val monitoringPeriods = getMonitoringPeriods(portfolio.startingMonitoringPeriod)
+
+                if ("eutaxonomy" in portfolio.monitoredFrameworks) {
+                    sendFinancialBulkDataRequest(enrichedPortfolio, monitoringPeriods)
+                    sendNonFinancialBulkDataRequest(enrichedPortfolio, monitoringPeriods)
+                    sendUndefinedBulkDataRequest(enrichedPortfolio, monitoringPeriods)
+                }
+
+                if ("sfdr" in portfolio.monitoredFrameworks) {
+                    sendSfdrBulkDataRequest(enrichedPortfolio, monitoringPeriods)
+                }
+            }
+        }
+
+        private fun getCompanyIdsBySector(
```

Comment:
```suggestion
        private fun filterCompanyIdsBySector(
```
as filterFunction is mandatory